mirrord-layer/src/go_hooks.rs

```diff
@@ -0,0 +1,429 @@
+#[cfg(target_os = "linux")]
+#[cfg(target_arch = "x86_64")]
+pub(crate) mod go_socket_hooks {
```

Comment:
I think the cfg is only needed in `lib.rs`, also probably don't need to mod here?

Reply:
if we don't do so, we get dead code warnings

Reply:
ok